pkg/secrets/aws_sm_client.go

```diff
@@ -0,0 +1,46 @@
+package secrets
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+)
+
+type AwsSecretsManagerClient interface {
+	FetchSecret(region string, secretName string) (*secretsmanager.GetSecretValueOutput, error)
+}
+
+type AwsSecretsManagerClientImpl struct{}
```

Comment:
Can you not have structs and interfaces named the same?

Reply:
![image](https://user-images.githubusercontent.com/711726/74866193-4d00dc80-5307-11ea-96d1-af409eec6dda.png)

☝️ Not according to clippy